src/queue.rs

```diff
@@ -239,7 +247,7 @@ impl<M: GuestAddressSpace> DescriptorChain<M> {
 
     /// Checks if this descriptor chain has another descriptor chain linked after it.
     pub fn has_next(&self) -> bool {
-        self.desc.flags & VIRTQ_DESC_F_NEXT != 0 && self.ttl > 1
+        self.has_next || self.curr_indirect.is_some()
```

Comment:
Is the `DescriptorChain::has_next` method really useful? It doesn't seem to guarantee that the following `next()` call will return something, and we can already look at the `Descriptor` object to determine if the chain is supposed to continue. I proposed removing it in #19 (would be great if you can have a look btw :D), and was wondering if there's a good reason to keep it around.

Reply:
I'd say the common use case for `DescriptorChain` is to simply iterate over it, so `DescriptorChain::has_next` may be redundant or even confusing, as it doesn't guarantee that `next()` will return something, as you said.

That said, if there's a use case for it (perhaps as a hint for an optimization?), I think we can keep it as long it's clearly stated in the documentation it's return value should only be used as a hint.

Reply:
I've started favoring the approach where only things known to be necessary get included, as new features can always be added later when they become relevant. I think this is one of those approaches that really pay off in the long run. At the same time, I'm by no means familiar with all the current use cases, so we can totally keep this if folks think it's worth it.